torch/_prims/__init__.py

```diff
@@ -1945,24 +1945,25 @@ def _convert_element_type_aten(a: Tensor, dtype: torch.dtype) -> Tensor:
 
 
 def _device_put_meta(
-    a: TensorLikeType, device: Union[str, torch.device]
+    a: TensorLikeType, device: Union[str, torch.device], non_blocking: bool
 ) -> TensorLikeType:
     assert isinstance(a, TensorLike)
     assert isinstance(device, (str, torch.device))
+    assert isinstance(non_blocking, bool)
 
     return TensorMeta(a, device=utils.canonicalize_device(device))
 
 
-def _device_put_aten(a: Tensor, device: Union[str, torch.device]) -> Tensor:
-    return a.to(device)
+def _device_put_aten(a: Tensor, device: Union[str, torch.device], non_blocking: bool) -> Tensor:
+    return a.to(device, non_blocking=non_blocking)
 
 
 _device_put_doc = """
   Creates a copy of a tensor on the given device.
   """
 
 device_put = _make_prim(
-    schema="device_put(Tensor a, Device device) -> Tensor",
+    schema="device_put(Tensor a, Device device, bool non_blocking) -> Tensor",
```

Comment:
Can we give this a default value of False for better backwards compatibility ? (even though its a bit surprising that is the default for pytorch).  I think you could also saw on failure - need to update https://github.com/pytorch/pytorch/blob/main/torch/_inductor/fx_passes/post_grad.py#L713-L715 with the new arg

Reply:
Would you also add the `non_blocking=True` arg to https://github.com/pytorch/pytorch/blob/main/torch/_inductor/fx_passes/post_grad.py#L713 ?

Reply:
To get a sense of which failing jobs are unrelated you can check: https://github.com/pytorch/pytorch/pull/136513#issuecomment-2370227076

Reply:
> Would you also add the `non_blocking=True` arg to https://github.com/pytorch/pytorch/blob/main/torch/_inductor/fx_passes/post_grad.py#L713 ?

Added it, althought it's not used anywhere in the function.

> To get a sense of which failing jobs are unrelated you can check: [#136513 (comment)](https://github.com/pytorch/pytorch/pull/136513#issuecomment-2370227076)

I could see that the labelling was failing, but I don't have the privileges to changes lables or run most actions.

> Added it. There are still some other tests failing due to NaN values (https://github.com/pytorch/pytorch/actions/runs/11007648290/job/30612720954), but I'm still looking into whether that's related or not.

Fixed this issue. I had created a bug in the C++ codegen. @eellison could you please re-run the tests? I think that should be all of them now.

Reply:
Thanks, it is pretty slow pipeline though. Regarding the failing lintrunners, not sure what those mean. If I run the failing command from this job (https://github.com/pytorch/pytorch/actions/runs/11074184476/job/30772425967) locally it seems to work fine:
![image](https://github.com/user-attachments/assets/8a1b3e5f-41ac-4ce3-a28d-4532788b5d9a)

Do you know what these jobs are supposed to do?

Reply:
Hi, to update lint, run `lintrunner init; lintrunner -a`

Reply:
@eellison fixed the linting issues from https://github.com/pytorch/pytorch/actions/runs/11079429031/job/30788465862. Please re-run CI.

The other blocker (https://github.com/pytorch/pytorch/actions/runs/11079429023/job/30788459451) appears to be an AWS EC2 issue so I'm assuming this is temporary.

Reply:
Fixed https://github.com/pytorch/pytorch/actions/runs/11085243879/job/30831253113 function signature issue from `mypy`. @eellison or @desertfire please re-run the CI.